app/services/concerns/xccdf/hosts.rb

```diff
@@ -28,16 +28,20 @@ def associate_rules_from_rule_results
       )
     end
 
+    def external_report?
+      test_result_profile.find_policy(account: @account, hosts: [@host]).nil?
```

Comment:
Observation: At this point, it might be good to think about having `find_policy` as a class method. No profile attribute (from `test_result_profile`) is used within the method.

```diff
diff --git a/app/models/concerns/profile_policy_association.rb b/app/models/concerns/profile_policy_association.rb
index 23a0b02..834568a 100644
--- a/app/models/concerns/profile_policy_association.rb
+++ b/app/models/concerns/profile_policy_association.rb
@@ -42,13 +42,17 @@ module ProfilePolicyAssociation
     # Lookup up internal profile that has the host(s)
     # assinged to a policy an which matches the ref_id
     # and ref_id of a benchmark (to ensure major OS version)
-    def find_policy(hosts: test_result_hosts, account: account_id)
+    def self.find_policy(hosts:, account:)
       Profile.includes(:benchmark, :policy_hosts)
              .where(policy_hosts: { host_id: hosts })
              .find_by(account: account, external: false, ref_id: ref_id,
                       benchmarks: { ref_id: benchmark.ref_id })
     end
 
+    def find_policy(hosts: test_result_hosts, account: account_id)
+      self.class.find_policy(hosts: hosts, account: account)
+    end
+
     def compliance_threshold
       policy_object&.compliance_threshold ||
         Policy::DEFAULT_COMPLIANCE_THRESHOLD
```

Reply:
The benchmark and profile ref_id are used from `self`.